src/Build/BuildCheck/Infrastructure/CheckWrapper.cs

```diff
@@ -52,10 +75,37 @@ internal void StartNewProject(
         }
     }
 
+    internal void ReportResult(BuildCheckResult result, ICheckContext checkContext, CheckConfigurationEffective config)
+    {
+        if (_reportsCountPerRule is not null)
+        {
```

Comment:
Let's limit the number of access to the dictionary.

```
if (!_reportsCountPerRule.TryGetValue(result.CheckRule.Id, out int currentCount))
{
    currentCount = 0;
}

if(currentCount >= MaxReportsNumberPerRule )
{
   return;
}

if(currentCount++ >= MaxReportsNumberPerRule )
{
   // ... Log and dismount
}

// ...

_reportsCountPerRule[result.CheckRule.Id] = currentCount;
```

Reply:
I can indeed limit accesses to the dictionary. If we want to dismount the check, we will need to limit the number of messages per check rather than per rule. Alternatively, we can have a more detailed check that all the rules reached the report limit and can produce no valuable info, which would be seldom case, and I think not worth code complication.